packages/machine/test/integration/test-commitment-store.ts

```diff
@@ -216,18 +205,15 @@ export class TestCommitmentStore {
     context: Context
   ): cf.legacy.node.ClientActionMessage | null {
     if (internalMessage.actionName === cf.legacy.node.ActionName.INSTALL) {
-      return getLastResult(Opcode.IO_WAIT, context.results).value;
+      return context.intermediateResults.inbox!;
     }
-    const incomingMessageResult = getLastResult(
-      Opcode.IO_WAIT,
-      context.results
-    );
-    if (JSON.stringify(incomingMessageResult) === JSON.stringify({})) {
+    const incomingMessageResult = context.intermediateResults.inbox!;
+    if (incomingMessageResult === undefined) {
```

Comment:
```suggestion
    if (typeof incomingMessageResult === "undefined") {
```

Reply:
@snario Let's not use `typeof` for this. If we need to check if `incomingMessageResult` isn't there, we either compare with `=== undefined` or the more relaxed `!incomingMessageResult`.